### DISCUSSION TITLE
Can a new version be released？

### DISCUSSION THREAD

**Discussion by wajncn** [quickfix-j/quickfixj#774] [Q&A] [answered]
The latest version is

```xml
<!-- https://mvnrepository.com/artifact/org.quickfixj/quickfixj-core -->
<dependency>
    <groupId>org.quickfixj</groupId>
    <artifactId>quickfixj-core</artifactId>
    <version>2.3.1</version>
</dependency>
```

But....    master branch many bugs have been fixed .     so, can you release a new version。

**Comment by chrjohn** [comment, marked as answer]
Please do not open issues for such general questions. Converting to discussion.

**Reply by chrjohn** [reply]
Currently still working on some issues for 3.0.0.
Have no ETA, sorry. Can't you build a version on your own?
Might have some overlaps with #769

**Reply by wajncn** [reply]
I think the master branch should be consistent with the released version

**Reply by McKratt** [reply]
Proxool is no more compatible with new Java version, we should have the new version with Hikari. Could I do something to speed up the next release ? Can I help in any way ?

**Reply by chrjohn** [reply]
Hi @McKratt , did you test 3.0.0-SNAPSHOT against your desired use cases and with your application and everything works?
Apart from that you could review pull requests for 3.0.0 issues or work on issues planned for 3.0.0 milestone. :)
Thank you :)

**Reply by McKratt** [reply]
Hi @chrjohn, sadly due to company policies, I cannot test the 3.0.0-SNAPSHOT. Except if you point to me a public Maven repository where it is already deployed.
I'll try to do what your propose thanks.

**Reply by chrjohn** [reply]
Hmm, that brings me directly to: https://github.com/quickfix-j/quickfixj/issues/769#issuecomment-2252776907 ;)

**Comment by dwlewi4** [comment]
It would be very useful to even tag a new release on top of the `QFJ_2_3_X` branch so we can update to `mina-core` 2.2.x

**Reply by chrjohn** [reply]
I will check if this could be done. However, it probably makes sense to cherry-pick at least: https://github.com/quickfix-j/quickfixj/pull/833
Need to check if other MINA-related PRs should also be cherry-picked.

**Reply by JamesTreasure** [reply]
Agreed would be very useful. We currently have a bug in QFJ (technically in MINA) and the problematic code looks to no longer exist in the newer versions of MINA so hoping this would fix it.
https://www.quickfixj.org/jira/browse/QFJ-997
https://issues.apache.org/jira/browse/DIRMINA-1159

**Reply by chrjohn** [reply]
@JamesTreasure are you sure that it no longer exists? Looking at https://github.com/apache/mina/blob/2.2.X/mina-core/src/main/java/org/apache/mina/proxy/filter/ProxyFilter.java I can still see the affected classes.

**Reply by JamesTreasure** [reply]
Ah my mistake, I meant I could no longer find it on trunk as the Proxy package doesn't even seem to exist anymore but guess that doesn't really help me.
https://github.com/apache/mina/tree/trunk/core/src/main/java/org/apache/mina

**Reply by chrjohn** [reply]
Trunk is not the version to be released. It is in the 2.2.x branch.
Would you be able to test if a new QFJ release with the current MINA fixes the issue?

**Reply by lopesgon** [reply]
Same need here! We had a bug while migrating to Java 17 and Kubernetes certainly due to Apache Mina dependency. We didn't not managed to confirm exactly where failure is occurring  but upgrading to the latest minor 2.1.x of mina-core fixed our issue.

It would be great to have a new release available instead of playing with dependency exclusion to fix it 😄

**Reply by chrjohn** [reply]
2.3.2 is available on Sonatype central now.

**Reply by MartyIX** [reply]
Thank you!!

**Comment by noblemaster** [comment]
Could this also be merged before a new release is built (appears to be ready)? MR: [Enhancement: First field in repeating group as delimiter](https://github.com/quickfix-j/quickfixj/pull/674#top)

Also yes, we would also like to have a new release. I can help out also if there is the need for additional help? Not sure how to go about it though?

**Reply by chrjohn** [reply]
Hi @noblemaster , problem with #674 is that it has conflicts since other PRs have been merged before. Especially #831 which introduced a new class `ValidationSettings`.
Help is always welcome, if you could look at PRs and review them that is always helpful.
Re new release: I doubt that 3.0.0 can be released this year due to other project work. A 2.3.2 release might work (but only with the MINA changes), but not sure at the moment. :-/

**Reply by noblemaster** [reply]
Let me discuss with my team: I should be able to re-do https://github.com/quickfix-j/quickfixj/pull/674 with the new `ValidationSettings`. Given it was last updated in 2023, I'll assume no one is working on it at the moment? Is that OK? I'll should be able to have it ready by end of next week.

For the 3.0.0, begin of next year (Jan. or Feb.) would be nice. Is that realistic?

**Reply by chrjohn** [reply]
OK, sounds good. Maybe it makes sense to ask the creator of that PR if he has the time to do it. Just to avoid that one of you spends their time in vain.
Edit: Feb sounds more realistic than Jan. ;) The current project has a timeframe until end of the year, so it should be easier after that.

**Reply by noblemaster** [reply]
February would be nice! I sent a message to the creator of the original PR. I'll try to get started next week unless I hear back... 👍

**Reply by x-leviathan** [reply]
Hi @chrjohn ,
I noticed that you mentioned the upcoming 2.3.2 release (mina changes). I would like to know if the mina changes will resolve the mina and proxy deadlock issue, as mentioned in #875 (if this conclusion is reliable) :)

Do you have an ETA for the 2.3.2 release? It would be helpful for our issue.

**Reply by chrjohn** [reply]
Hi @x-leviathan , no currently don't have an ETA apart from "probably in the next two months". ;)
To verify it it fixes that bug I would need help from someone who can test it against a proxy. I unfortunately cannot.
